src/diffpy/srmise/tests/test_dataclusters.py

```diff
@@ -32,3 +33,86 @@ def test___eq__():
             print(f"not-equal test failed on {attr_key}")
             assert not expected == actual
         attributes.update({attr_key: reset})
+
+    # In the set data test, we test for these cases.
+    # (1) x and y are non-empty array values, and res is positive (the most generic case)
+    # (2) x and y are non-empty array values, and res is 0 (will produce a msg that makes trivial clustering)
+    # (3) x and y are non-empty array values, and res is negative (will produce a ValueError,
+    # msg = please enter a non-negative res value)
+    # (4, 5) One of x and y is empty array, and res is positive
```

Comment:
Here I woud test for len(x) != len(y) (ValueError) and not worry about the other cases.  The user won't give a len(0) array on purpose and the code will error in that case anyway (on sort for example) in case the user does it inadvertently.